### PR TITLE
UX: Fix Validation and Error Handling Issues in Registration and Login Form

### PR DESCRIPTION
**Pull request by eleonoreeuzenes**
### Pull Request Description

This pull request addresses and fixes issues related to validation and error handling in the `RegisterView` and `LoginView` registration form. Previously, validation errors were being displayed prematurely, before the user had finished typing. The changes made ensure that validation errors are only shown after the user has finished interacting with each field.

**Changes Made:**

1. **Fixed Validation Timing:**
   - **Username Field**: Adjusted the validation to trigger only after the user has finished typing and the field loses focus. Ensured that errors are displayed only when the input does not meet the required criteria.
   - **Email Field**: Corrected validation to properly validate email addresses after the field loses focus, preventing premature error messages.
   - **Password Field**: Fixed validation logic to check for password strength only when the user exits the password field, ensuring accurate error reporting.
   - **Repeated Password Field**: Updated validation to ensure that error messages for mismatched passwords are shown only after the field loses focus.

2. **Improved Error Handling:**
   - **Error Message Display**: Implemented proper logic to show error messages under the respective input fields only after validation is triggered. This prevents errors from being shown while the user is still typing.

3. **FocusNode Management:**
   - Adjusted `FocusNode` listeners to correctly handle focus changes and trigger validation only when necessary, ensuring a smoother user experience.

4. **UI and UX Enhancements:**
   - Ensured that error messages are updated and cleared appropriately based on user input, improving the overall user experience.

5. **Code Refactoring:**
   - Cleaned up validation logic and removed redundant listeners to streamline code and enhance readability.

**Testing:**

- Confirmed that validation errors are now only displayed after the user finishes typing and navigates away from the field.
- Validated that error messages are correctly shown and cleared as per user input.
- Conducted tests to ensure that the form behaves as expected across different scenarios and edge cases.

**Related Issues:**

- Fixes [#93](https://github.com/Eco-Gest/Frontend-Flutter/issues/93)

### CODE REVIEW

File: lib/views/auth/test.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
```
Comment (AugustinSeguin) [inline lib/views/auth/test.dart:7]: à supprimer je pense :D

**Review by AugustinSeguin — CHANGES_REQUESTED** [review]

**Review by AugustinSeguin — APPROVED** [review]
J'ai modifié le message d'erreur du mot de passe : "minimum 8 caractères" et j'ai supprimé le fichier de test
